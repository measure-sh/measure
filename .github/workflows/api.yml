name: API
on:
  push:
    branches:
      - 'main'
    paths:
      - .github/workflows/api.yml
      - backend/api/dockerfile
      - backend/api/**/*.go
      - backend/api/**/*.mod
      - backend/api/**/*.sum
  pull_request:
    paths:
      - .github/workflows/api.yml
      - backend/api/dockerfile
      - backend/api/**/*.go
      - backend/api/**/*.mod
      - backend/api/**/*.sum

env:
  REGISTRY: ghcr.io
  IMAGE_NAME: measure-sh/api

jobs:
  build:
    name: Build
    runs-on: ubuntu-latest
    strategy:
      matrix:
        go-version: ["1.22.x"]
    defaults:
      run:
        working-directory: backend/api
    steps:
      - name: Checkout repository
        uses: actions/checkout@v4
      - name: Setup go
        uses: actions/setup-go@v5
        with:
          go-version: ${{ matrix.go-version }}
          check-latest: true
          cache-dependency-path: backend/api/go.sum
      - name: Display go version
        run: go version
      - name: Install dependencies
        run: go get .
      - name: Build with ${{ matrix.go-version }}
        run: go build -v ./...

  test:
    name: Test
    runs-on: ubuntu-latest
    strategy:
      matrix:
        go-version: ["1.22.x"]
    defaults:
      run:
        working-directory: backend/api
    steps:
      - name: Checkout repository
        uses: actions/checkout@v4
      - name: Setup go
        uses: actions/setup-go@v5
        with:
          go-version: ${{ matrix.go-version }}
          check-latest: true
          cache-dependency-path: backend/api/go.sum
      - name: Display go version
        run: go version
      - name: Install dependencies
        run: go get .
      - name: Test with ${{ matrix.go-version }}
        run: go test ./...
  
  push:
    name: Build and push image
    runs-on: ubuntu-latest
    if: github.ref_type == 'tag' && startsWith(github.ref, 'v')
    needs: [build, test]
    permissions:
      contents: read
      packages: write
      attestations: write
      id-token: write
    steps:
      - name: Checkout repository
        uses: actions/checkout@v4

      - name: Login to Container Registry
        uses: docker/login-action@v3
        with:
          registry: ${{ env.REGISTRY }}
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}

      - name: Extract metadata (tags, labels) for Docker
        id: meta
        uses: docker/metadata-action@v5
        with:
          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}

      - name: Build and push OCI image
        id: push
        uses: docker/build-push-action@v6
        with:
          context: backend/api
          file: backend/api/dockerfile
          push: true
          tags: ${{ steps.meta.outputs.tags }}
          labels: |
            org.opencontainers.image.source=https://github.com/measure-sh/measure/tree/${{ github.ref_name }}/backend/api/dockerfile
            org.opencontainers.image.description=Measure API Server
            org.opencontainers.image.licenses=Apache-2.0

      # Disable artifact attestation till repo is public
      # as this is a paid feature in GitHub.
      # - name: Generate artifact attestation
      #   uses: actions/attest-build-provenance@v1
      #   with:
      #     subject-name: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
      #     subject-digest: ${{ steps.push.outputs.digest }}
      #     push-to-registry: true

  deploy:
    name: Deploy to fly.io
    if: github.ref == 'refs/heads/main'
    needs: [build, test]
    runs-on: ubuntu-latest
    defaults:
      run:
        working-directory: backend/api
    steps:
      - name: Checkout repository
        uses: actions/checkout@v4
      - name: Setup flyctl
        uses: superfly/flyctl-actions/setup-flyctl@master
      - name: Deploy to fly.io
        run: flyctl deploy --remote-only
        env:
          FLY_API_TOKEN: ${{ secrets.FLY_API_TOKEN }}
