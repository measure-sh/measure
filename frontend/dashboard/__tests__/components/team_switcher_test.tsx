import '@testing-library/jest-dom'
import { render, screen, fireEvent } from '@testing-library/react'
import { expect, describe, it } from '@jest/globals'
import TeamSwitcher, { TeamsSwitcherStatus } from '@/app/components/team_switcher'
import { Team } from '@/app/api/api_calls'

describe('TeamSwitcher', () => {
    it('renders correctly in collapsed state', () => {
        const items = [{ id: '1', name: 'Team 1' }, { id: '2', name: 'Team 2' }, { id: '3', name: 'Team 3' }]

        const container = render(<TeamSwitcher items={items} teamsSwitcherStatus={TeamsSwitcherStatus.Success} onChangeSelectedItem={() => { }} />)

        expect(container).toMatchSnapshot()
    })

    it('renders correctly on passing in initial item index', () => {
        const items = [{ id: '1', name: 'Team 1' }, { id: '2', name: 'Team 2' }, { id: '3', name: 'Team 3' }]

        const container = render(<TeamSwitcher items={items} initialItemIndex={2} teamsSwitcherStatus={TeamsSwitcherStatus.Success} onChangeSelectedItem={() => { }} />)

        expect(container).toMatchSnapshot()
    })

    it('renders correctly in loading state', () => {
        const items = [{ id: '1', name: 'Team 1' }, { id: '2', name: 'Team 2' }, { id: '3', name: 'Team 3' }]

        const container = render(<TeamSwitcher items={items} initialItemIndex={2} teamsSwitcherStatus={TeamsSwitcherStatus.Loading} onChangeSelectedItem={() => { }} />)

        expect(container).toMatchSnapshot()
    })

    it('renders correctly in error state', () => {
        const items = [{ id: '1', name: 'Team 1' }, { id: '2', name: 'Team 2' }, { id: '3', name: 'Team 3' }]

        const container = render(<TeamSwitcher items={items} initialItemIndex={2} teamsSwitcherStatus={TeamsSwitcherStatus.Error} onChangeSelectedItem={() => { }} />)

        expect(container).toMatchSnapshot()
    })

    it('renders correctly in opened state', () => {
        const items = [{ id: '1', name: 'Team 1' }, { id: '2', name: 'Team 2' }, { id: '3', name: 'Team 3' }]

        const container = render(<TeamSwitcher items={items} teamsSwitcherStatus={TeamsSwitcherStatus.Success} onChangeSelectedItem={() => { }} />)

        expect(container).toMatchSnapshot()

        fireEvent.click(screen.getByText('Team 1'))

        expect(container).toMatchSnapshot()
    })

    it('renders correctly on new team selection', () => {
        const items = [{ id: '1', name: 'Team 1' }, { id: '2', name: 'Team 2' }, { id: '3', name: 'Team 3' }]

        const container = render(<TeamSwitcher items={items} teamsSwitcherStatus={TeamsSwitcherStatus.Success} onChangeSelectedItem={() => { }} />)

        expect(container).toMatchSnapshot()

        fireEvent.click(screen.getByText('Team 1'))

        fireEvent.click(screen.getByText('Team 3'))

        expect(container).toMatchSnapshot()
    })

    it('calls listener new team selection', () => {
        const items = [{ id: '1', name: 'Team 1' }, { id: '2', name: 'Team 2' }, { id: '3', name: 'Team 3' }]

        let newSelectedItem = {} as Team
        let onChangeSelectedItem = (item: Team) => newSelectedItem = item

        render(<TeamSwitcher items={items} teamsSwitcherStatus={TeamsSwitcherStatus.Success} onChangeSelectedItem={onChangeSelectedItem} />)

        fireEvent.click(screen.getByText('Team 1'))

        fireEvent.click(screen.getByText('Team 3'))

        expect(newSelectedItem).toStrictEqual({ id: '3', name: 'Team 3' })

    })
})