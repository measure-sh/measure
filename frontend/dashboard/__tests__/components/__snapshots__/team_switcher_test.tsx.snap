// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamSwitcher renders correctly in collapsed state 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-50 relative w-40 self-center inline-block text-left"
      >
        <button
          class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
          type="button"
        >
          <div
            class="flex flex-row justify-center"
          >
            <p
              class="pl-8 truncate w-max"
            >
              Team 1
            </p>
            <p
              class="pl-3 pr-4 pt-1 text-sm"
            >
              ⏷
            </p>
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-50 relative w-40 self-center inline-block text-left"
    >
      <button
        class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
        type="button"
      >
        <div
          class="flex flex-row justify-center"
        >
          <p
            class="pl-8 truncate w-max"
          >
            Team 1
          </p>
          <p
            class="pl-3 pr-4 pt-1 text-sm"
          >
            ⏷
          </p>
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TeamSwitcher renders correctly in error state 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-50 relative w-40 self-center inline-block text-left"
      >
        <button
          class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
          disabled=""
          type="button"
        >
          <p
            class="pl-8 truncate w-max"
          >
            Error
          </p>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-50 relative w-40 self-center inline-block text-left"
    >
      <button
        class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
        disabled=""
        type="button"
      >
        <p
          class="pl-8 truncate w-max"
        >
          Error
        </p>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TeamSwitcher renders correctly in loading state 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-50 relative w-40 self-center inline-block text-left"
      >
        <button
          class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
          disabled=""
          type="button"
        >
          <p
            class="pl-8 truncate w-max"
          >
            Updating...
          </p>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-50 relative w-40 self-center inline-block text-left"
    >
      <button
        class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
        disabled=""
        type="button"
      >
        <p
          class="pl-8 truncate w-max"
        >
          Updating...
        </p>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TeamSwitcher renders correctly in opened state 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-50 relative w-40 self-center inline-block text-left"
      >
        <button
          class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
          type="button"
        >
          <div
            class="flex flex-row justify-center"
          >
            <p
              class="pl-8 truncate w-max"
            >
              Team 1
            </p>
            <p
              class="pl-3 pr-4 pt-1 text-sm"
            >
              ⏷
            </p>
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-50 relative w-40 self-center inline-block text-left"
    >
      <button
        class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
        type="button"
      >
        <div
          class="flex flex-row justify-center"
        >
          <p
            class="pl-8 truncate w-max"
          >
            Team 1
          </p>
          <p
            class="pl-3 pr-4 pt-1 text-sm"
          >
            ⏷
          </p>
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TeamSwitcher renders correctly in opened state 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-50 relative w-40 self-center inline-block text-left"
      >
        <button
          class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
          type="button"
        >
          <div
            class="flex flex-row justify-center"
          >
            <p
              class="pl-8 truncate w-max"
            >
              Team 1
            </p>
            <p
              class="pl-3 pr-4 pt-1 text-sm"
            >
              ⏷
            </p>
          </div>
        </button>
        <div
          class="origin-top-right absolute left-0 mt-2 w-48 rounded-md shadow-lg ring-1 ring-black ring-opacity-5"
        >
          <div
            aria-labelledby="options-menu"
            aria-orientation="vertical"
            role="menu"
          >
            <button
              class="block w-full px-2 py-2 text-white bg-neutral-950 font-display text-left hover:text-black hover:bg-yellow-200 active:bg-yellow-300 outline-hidden focus:bg-yellow-200"
              role="menuitem"
            >
              Team 1
            </button>
            <button
              class="block w-full px-2 py-2 text-white bg-neutral-950 font-display text-left hover:text-black hover:bg-yellow-200 active:bg-yellow-300 outline-hidden focus:bg-yellow-200"
              role="menuitem"
            >
              Team 2
            </button>
            <button
              class="block w-full px-2 py-2 text-white bg-neutral-950 font-display text-left hover:text-black hover:bg-yellow-200 active:bg-yellow-300 outline-hidden focus:bg-yellow-200"
              role="menuitem"
            >
              Team 3
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-50 relative w-40 self-center inline-block text-left"
    >
      <button
        class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
        type="button"
      >
        <div
          class="flex flex-row justify-center"
        >
          <p
            class="pl-8 truncate w-max"
          >
            Team 1
          </p>
          <p
            class="pl-3 pr-4 pt-1 text-sm"
          >
            ⏷
          </p>
        </div>
      </button>
      <div
        class="origin-top-right absolute left-0 mt-2 w-48 rounded-md shadow-lg ring-1 ring-black ring-opacity-5"
      >
        <div
          aria-labelledby="options-menu"
          aria-orientation="vertical"
          role="menu"
        >
          <button
            class="block w-full px-2 py-2 text-white bg-neutral-950 font-display text-left hover:text-black hover:bg-yellow-200 active:bg-yellow-300 outline-hidden focus:bg-yellow-200"
            role="menuitem"
          >
            Team 1
          </button>
          <button
            class="block w-full px-2 py-2 text-white bg-neutral-950 font-display text-left hover:text-black hover:bg-yellow-200 active:bg-yellow-300 outline-hidden focus:bg-yellow-200"
            role="menuitem"
          >
            Team 2
          </button>
          <button
            class="block w-full px-2 py-2 text-white bg-neutral-950 font-display text-left hover:text-black hover:bg-yellow-200 active:bg-yellow-300 outline-hidden focus:bg-yellow-200"
            role="menuitem"
          >
            Team 3
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TeamSwitcher renders correctly on new team selection 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-50 relative w-40 self-center inline-block text-left"
      >
        <button
          class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
          type="button"
        >
          <div
            class="flex flex-row justify-center"
          >
            <p
              class="pl-8 truncate w-max"
            >
              Team 1
            </p>
            <p
              class="pl-3 pr-4 pt-1 text-sm"
            >
              ⏷
            </p>
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-50 relative w-40 self-center inline-block text-left"
    >
      <button
        class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
        type="button"
      >
        <div
          class="flex flex-row justify-center"
        >
          <p
            class="pl-8 truncate w-max"
          >
            Team 1
          </p>
          <p
            class="pl-3 pr-4 pt-1 text-sm"
          >
            ⏷
          </p>
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TeamSwitcher renders correctly on new team selection 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-50 relative w-40 self-center inline-block text-left"
      >
        <button
          class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
          type="button"
        >
          <div
            class="flex flex-row justify-center"
          >
            <p
              class="pl-8 truncate w-max"
            >
              Team 3
            </p>
            <p
              class="pl-3 pr-4 pt-1 text-sm"
            >
              ⏷
            </p>
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-50 relative w-40 self-center inline-block text-left"
    >
      <button
        class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
        type="button"
      >
        <div
          class="flex flex-row justify-center"
        >
          <p
            class="pl-8 truncate w-max"
          >
            Team 3
          </p>
          <p
            class="pl-3 pr-4 pt-1 text-sm"
          >
            ⏷
          </p>
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TeamSwitcher renders correctly on passing in initial item index 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-50 relative w-40 self-center inline-block text-left"
      >
        <button
          class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
          type="button"
        >
          <div
            class="flex flex-row justify-center"
          >
            <p
              class="pl-8 truncate w-max"
            >
              Team 3
            </p>
            <p
              class="pl-3 pr-4 pt-1 text-sm"
            >
              ⏷
            </p>
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-50 relative w-40 self-center inline-block text-left"
    >
      <button
        class="aspect-square w-full text-xl font-display border border-black rounded-full outline-hidden hover:enabled:bg-yellow-200 focus:enabled:bg-yellow-200 active:enabled:bg-yellow-300"
        type="button"
      >
        <div
          class="flex flex-row justify-center"
        >
          <p
            class="pl-8 truncate w-max"
          >
            Team 3
          </p>
          <p
            class="pl-3 pr-4 pt-1 text-sm"
          >
            ⏷
          </p>
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
